### ISSUE TITLE
cannot build rpm due to L10N issues

### ISSUE THREAD

**Issue by jrgm** [mozilla/persona#3683]
The train-2013.07.17 branch fails at building the rpm. 

I see @shane-tomlinson put in a fix in https://github.com/mozilla/browserid/pull/3673 which is in train-2013.07.17 for pushing to an awsbox, but when building the rpm with a full complement of L10N translation from `verbatim`, the build fails with errors like:

```
...
...
built /production/ro/browserid.css in 0.70s
built /production/sk/browserid.css in 0.71s
failed to build /production/ca/dialog.js : { errno: 34,
  code: 'ENOENT',
  path: '/private/tmp/build_this_thing/browserid/resources/static/i18n/ca/client.js' }
failed to build /production/af/dialog.js : { errno: 34,
  code: 'ENOENT',
  path: '/private/tmp/build_this_thing/browserid/resources/static/i18n/af/client.js' }
failed to build /production/cs/dialog.js : { errno: 34,
  code: 'ENOENT',
  path: '/private/tmp/build_this_thing/browserid/resources/static/i18n/cs/client.js' }
...
   etc., 
...
```

To decouple the problem from `rpm`, this script on `osx` reproduces the error. Probably need some changes in `./locale/compile-json.sh` from the SVN repo for browserid. 

```
#!/bin/sh

set -x
TEST_DIR=/tmp/build_this_thing

/bin/rm -rf $TEST_DIR
mkdir -p $TEST_DIR
cd $TEST_DIR

git clone https://github.com/mozilla/browserid.git $TEST_DIR/browserid
cd $TEST_DIR/browserid
svn co http://svn.mozilla.org/projects/l10n-misc/trunk/browserid/locale
git checkout -b train-2013.07.17

npm install

export PATH=$PWD/node_modules/.bin:$PATH

sh -x ./locale/compile-json.sh locale/ resources/static/i18n/

env CONFIG_FILES=$PWD/config/l10n-all.json scripts/compress
```

Since this blocks the train, marking 5-star. /cc @ozten

**Comment by ozten** [comment]
I think we need to switch from scripts checked into svn and start using scripts from i18n-abide.

Does this create a chicken and egg problem, since `node_modules/i18n-abide/bin` won't exist during rpm generation?

I'll jump on this first thing tomorrow.

**Comment by jrgm** [comment]
Offhand, I think `node_modules/i18n-abide/bin` will be there when scripts/browserid.spec does its stuff.

**Comment by ozten** [comment]
I think we can replace 
    ./locale/compile-json.sh locale/ resources/static/i18n/

with
    ./node_modules/.bin/compile-json locale/ resources/static/i18n/

in `scripts/browserid.spec`.

## more background

A possible improvement is to make this closer to what we do in AWS boxes and locally on dev envs.

So `scripts/l10n-update.js` is what pulls SVN and calls `compile-json`. It creates `client.js`, `client.json`, and the same pair for messages.

Before there was only `client.json` which was actually a JavaScript file. Now, with i18n-abide 0.0.8, we create the two files and `client.json` is a valid JSON file.

`scripts/l10n-update.js` isn't currently used to build the rpms, which is a step I missed. We could replace calling 

```
./locale/compile-json.sh locale/ resources/static/i18n/
```

with a call to `scripts/l10n-update.js`. The side effects are that this script also updates or creates the `locale` directory.

**Comment by ozten** [comment]
We can close this, right?

**Comment by jrgm** [comment]
Yes, although that rpm has not yet been deployed to an aws stack.

**Comment by jrgm** [comment]
Closing.

**Comment by jrgm** [comment]
I shouldn't have closed this, because this needs the "longer-term" fix.

I can't build the train-2013.07.31 rpm at the moment, so I will cherrypick the fix made to train-2013.07.17 onto the 07.31 train.

**Comment by jrgm** [comment]
Still (probably) needs a fix for the train-2013.08.14 to be branched Wednesday.

**Comment by ozten** [comment]
Okay, this adds it to dev. Would be great to merge this before cutting the next train.
https://github.com/mozilla/browserid/pull/3788

Thanks @jrgm
